docs/utils/download.rst

```diff
@@ -0,0 +1,43 @@
+.. _pyvo-download:
+
+**************************************************
+Download Utilities (`pyvo.utils.download`)
+**************************************************
+
+The download utilities provide methods for downloading data once a link to it
+it obtained. These can be considered an advanced version of `~pyvo.dal.Record.getdataset` that can handle
+data from standard on-prem servers as well as cloud data. For now only AWS is supported.
+
+There two methods with the same call signature: `http_download` and `aws_download`. The first handles
```

Comment:
use full API links otherwise sphinx will complain (we don't have nitpicky turned on yet, but things like this will eventually cause CI failures)